src/main/java/io/nitric/faas/Faas.java

```diff
@@ -65,86 +64,106 @@
      */
-    public Faas port(int port) {
-        this.port = port;
+    protected Faas stub(FaasGrpc.FaasStub stub) {
+        this.stub = stub;
         return this;
     }
 
+    // Public Methods -------------------------------------------------------------------
     /**
      * Start the FaaS server after configuring the given function.
+     * This method will block until the stream has terminated
      *
      * @param function the function (required)
      */
     public void startFunction(NitricFunction function) {
         Objects.requireNonNull(function, "null function parameter");
 
-        if (httpServer != null) {
-            throw new IllegalStateException("server already started");
+        // FIXME: Uncoverable code without mocking static methods (need to include Powermock)
+        // Once we've asserted that this interfaces with a mocked stream observer
+        // We only need to assert the FaasGrpc.newStub is called to cover this logic in
+        // the case where the user has not provided a custom stub
+        if (this.stub == null) {
+            // Create a default stub with the singleton channel
+            // TODO: Determine if we should use a dedicated channel for this FaaS loop?
+            this.stub = FaasGrpc.newStub(GrpcChannelProvider.getChannel());
         }
 
-        var childAddress = System.getenv("CHILD_ADDRESS");
-        if (childAddress != null && !childAddress.isBlank()) {
-            hostname = childAddress;
-        }
-
-        try {
-            httpServer = HttpServer.create(new InetSocketAddress(hostname, port), 0);
-
-            httpServer.createContext("/", buildServerHandler(function));
-
-            httpServer.setExecutor(null);
-
-            // Start the server
-            httpServer.start();
-
-            var builder = new StringBuilder().append(getClass().getSimpleName());
-            if (DEFAULT_HOSTNAME.equals(hostname)) {
-                builder.append(" listening on port ").append(port);
-
-            } else {
-                builder.append(" listening on ").append(hostname).append(":").append(port);
+        AtomicReference<StreamObserver<ClientMessage>> clientObserver = new AtomicReference<>();
+        // Add a latch to block on while the stream is running
+        CountDownLatch finishedLatch = new CountDownLatch(1);
+        // Begin the stream
+        var observer = this.stub.triggerStream(new StreamObserver<>() {
+            @Override
+            public void onNext(ServerMessage serverMessage) {
+                // We got a new message from the server
+                switch (serverMessage.getContentCase()) {
+                    case INIT_RESPONSE:
+                        // We have an init ack from the membrane
+                        // XXX: NO OP for now
+                        break;
+                    case TRIGGER_REQUEST:
+                        var trigger = Trigger.buildTrigger(serverMessage.getTriggerRequest());
```

Comment:
Could you add some try catch error handling around the user land function logging contextual information for developer debugging.

Reply:
No problem, wanted to ask what your preferred logging method was as well, can add stdio logging, but would of course would be much better to use something more structured.,

Reply:
At the moment just use standard error. Thanks